### ISSUE TITLE
Verification of signed messages 

### ISSUE THREAD

**Issue by jmn** [isaacs/github#865]
I currently don't have a strong use case for this but as I created [another issue re. messages](https://github.com/isaacs/github/issues/864), I also thought about the possibility of performing GPG message verification in messages in the Github user interface. This would serve to ensure that no one could make claims in your name unless security is compromised.

**Comment by cirosantilli** [comment]
But how would that work? To GPG sign automatically you'd need to give your private keys to GitHub.

**Comment by jmn** [comment]
Github wouldn't provide signing facilities, that would be the
responsibility of the user (e.g. cut and paste signed message or use a
browser extension.) Github would provide a verification interface which
shows whether a signed message is valid.

**Comment by aspiers** [comment]
I'm not convinced the cut+paste or browser extension ideas would be workable.  However, let's take a step back and try to get to the bottom of what this feature request is really about...

I think the key part of "This would serve to ensure that no one could make claims in your name unless security is compromised" is the "in your name".  What does "name" actually mean here?  It means an identity.  But since we're online, that means a digital identity, and that can take many forms, including a GitHub user and the owner of a GPG private key.  (I think we can overlook the "unless security is compromised" clause, since that problem exists for *any* form of digital authentication, not just a GitHub account or GPG private key.)

Since you can only post comments in GitHub when you are already authenticated, clearly the comments you posted are already verifiably linked to your GitHub account, simply by looking at GitHub's UI or API.  What you seem to be requesting is that this GitHub identity can also be verifiably associated with the owner of a GPG private key.  And the good news is that that is already possible :-)  This is exactly what https://keybase.io/ does.  For example, if you look at https://keybase.io/adamspiers, you can see that the GitHub account I am using right now is associated with GPG public key D31B 5563 DAC1 D4FA in a cryptographically secure and independently verifiable way.

So in summary, I'd suggest that what you want is already achieved very nicely by keybase.io, so you should just use that :)

**Comment by jmn** [comment]
Yes I think there is a bunch of things that Github can think about improving when it comes to protecting digital identites. Perhaps GPG is not the right tool for messages, but I do think that it adds another level of security and could be useful in some critical situtations - but unless someones can show they actually need it I don't think this should be given much effort.

Another example where I think this could be improved is that it would be more visually apparent in the UI when someone apart from the user listed as the message owner has edited a message. Right now it only shows on mouse-over/hover tooltip on "edited". And there is no edit history. Which has its pros and cons.
